docs/components/contacts.rst

```diff
@@ -1,4 +1,257 @@
+            // return value based on the data included in getQueryOptions() if used in the stats method above.
+            $event->addToCounter($eventTypeKey, $stats);
+
+            if (!$event->isEngagementCount()) {
+                // Add the events to the event array
+                foreach ($stats['results'] as $stat) {
+                    if ($stat['dateSent']) {
+                        $event->addEvent(
+                            [
+                                // Event key type
+                                'event'           => $eventTypeKey,
+                                // Event name/label - can be a string or an array as below to convert to a link
+                                'eventLabel'      => [
+                                    'label' => $stat['name'],
+                                    'href'  => $this->router->generate(
+                                        'mautic_dynamicContent_action',
+                                        ['objectId' => $stat['dynamic_content_id'], 'objectAction' => 'view']
+                                    )
+                                ],
+                                // Translated string displayed in the Event Type column
+                                'eventType'       => $eventTypeName,
+                                // \DateTime object for the timestamp column
+                                'timestamp'       => $stat['dateSent'],
+                                // Optional details passed through to the contentTemplate
+                                'extra'           => [
+                                    'stat' => $stat,
+                                    'type' => 'sent'
+                                ],
+                                // Optional template to customize the details of the event in the timeline
+                                'contentTemplate' => 'MauticDynamicContentBundle:SubscribedEvents\Timeline:index.html.php',
+                                // Font Awesome class to display as the icon
+                                'icon'            => 'fa-envelope'
+                            ]
+                        );
+                    }
+                }
+            }
+        }
+    }
+
+.. list-table::
+    :header-rows: 1
+
+    * - Method
+      - Description
+    * - ``isApplicable()``
+      - Determines if this event is applicable and not filtered out.
+    * - ``addEventType()``
+      - Required - Add this event to the list of available events.
+    * - ``getLead()``
+      - Get the Contact entity
+    * - ``getQueryOptions()``
+      - Used to get pagination, filters, etc needed to generate an appropriate query.
+    * - ``addToCounter()``
+      - Used to add total number of events across all Pages to the counters. This also generates the numbers for the engagements graph.
+    * - ``addEvent()``
+      - Required - Injects an event into the timeline. Accepts an array with the keys defined as below. 
```

Comment:
This was in the original docs, but folks can definitely get this from the `LeadTimelineEvent` as well - long-term, I'd actually be in favor of adding this info to the event itself and have auto-generated docs on methods. Should be much less of a maintenance burden then